CMakeLists.txt

```diff
@@ -0,0 +1,203 @@
+    NO_DEFAULT_PATH)
+  find_library(FONTCONFIG_LIBRARY fontconfig)
+
+  if (FONTCONFIG_LIBRARY)
+    message(STATUS "Found Fontconfig: ${FONTCONFIG_LIBRARY}")
+  else()
+    message(WARNING "fontconfig not found. Cannot use freetype!")
+  endif (FONTCONFIG_LIBRARY)
+  
+  if (FREETYPE_FOUND AND FONTCONFIG_LIBRARY)
+
+    list(APPEND _glvis_compile_defs "-DGLVIS_USE_FREETYPE")
+
+    list(APPEND _glvis_include_dirs "${FREETYPE_INCLUDE_DIRS}")
+    
+    list(APPEND _glvis_library_dirs "${FREETYPE_LIBRARIES}")
+    list(APPEND _glvis_library_dirs "${FONTCONFIG_LIBRARY}")
+
+  else()
+    set(GLVIS_USE_FREETYPE OFF)
+  endif (FREETYPE_FOUND AND FONTCONFIG_LIBRARY)
+endif (GLVIS_USE_FREETYPE)
+
+# FIXME: MFEM exports the MPI include directory in its
+# MFEM_INCLUDE_DIRS variable. However, I don't see it exporting the
+# MPI libraries, so we need to go find them. For now, I use the
+# include path to verify that the MPI library matches.
+if (MFEM_USE_MPI)
+  find_package(MPI REQUIRED)
+  find_package(LAPACK REQUIRED)
+  
+  list(FIND MFEM_INCLUDE_DIRS ${MPI_INCLUDE_PATH} _found_correct_mpi)
+  if (_found_correct_mpi EQUAL -1)
+    message(FATAL_ERROR
+      "Found a different version of MPI than used to build MFEM.")
+  endif (_found_correct_mpi EQUAL -1)
+
+  # Sometimes these cause problems...
+  set(_mpi_compile_flags)
+  set(_mpi_link_flags)
+  if (UNIX)
+    separate_arguments(
+      _mpi_compile_flags UNIX_COMMAND "${MPI_CXX_COMPILE_FLAGS}")
+    separate_arguments(
+      _mpi_link_flags UNIX_COMMAND "${MPI_CXX_LINK_FLAGS}")
+  else()
+    separate_arguments(
+      _mpi_compile_flags WINDOWS_COMMAND "${MPI_CXX_COMPILE_FLAGS}")
+    separate_arguments(
+      _mpi_link_flags WINDOWS_COMMAND "${MPI_CXX_LINK_FLAGS}")
+  endif (UNIX)
+  
+  list(APPEND _glvis_compiler_opts "${_mpi_compile_flags}")
+  list(APPEND _glvis_libraries "${_mpi_link_flags}")
+  list(APPEND _glvis_libraries "${MPI_CXX_LIBRARIES}")
+  list(APPEND _glvis_include_dirs "${MPI_CXX_INCLUDE_PATH}")
+
+  list(APPEND _glvis_libraries "${LAPACK_LINKER_FLAGS}")
+  list(APPEND _glvis_libraries "${LAPACK_LIBRARIES}")
+endif (MFEM_USE_MPI)
+
+# Find threading library
+if (GLVIS_USE_THREADS)
+  find_package(Threads)
+  if (NOT THREADS_FOUND)
+    message(WARNING "No threading model found. Threads disabled.")
+    set(GLVIS_USE_THREADS OFF)
+  else()
+    list(APPEND _glvis_libraries "${CMAKE_THREAD_LIBS_INIT}")
+  endif (NOT THREADS_FOUND)
+endif (GLVIS_USE_THREADS)
+
+
+#
+# Setup the GLVis library target
+#
+
+add_subdirectory(lib)
+
+#
+# Setup the GLVis executable
+
+add_executable(glvis-exe glvis.cpp)
+set_target_properties(glvis-exe PROPERTIES OUTPUT_NAME glvis)
+
+target_link_libraries(glvis-exe PRIVATE glvis)
+
+# Install the executable
+install(TARGETS glvis-exe RUNTIME DESTINATION bin)
+
+# Install the gnutls helper script
+if (MFEM_USE_GNUTLS)
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/glvis-keygen.sh
+    DESTINATION bin
```

Comment:
This matches the destination (sort of) from the makefile. The makefile installs things to ```${PREFIX}``` and here we install things to ```${PREFIX}/bin```, which is more consistent with other (e.g., autotools) install paths. Do these really belong together? Perhaps ```${PREFIX}/share```?